apps/ledger-live-mobile/src/screens/SyncOnboarding/ResyncOverlay.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState } from "react";
+import { Flex, InfiniteLoader, Text } from "@ledgerhq/native-ui";
+import { useTheme } from "styled-components/native";
+import { useTranslation } from "react-i18next";
+
+type Props = {
+  isOpen: boolean;
+  productName: string;
+  delay?: number;
+};
+
+const ResyncOverlay = ({ isOpen, delay = 0, productName }: Props) => {
+  const { t } = useTranslation();
+  const [showContent, setShowContent] = useState<boolean>(false);
+  const { colors, radii } = useTheme();
+
+  useEffect(() => {
+    if (isOpen) {
+      setTimeout(() => {
+        setShowContent(true);
+      }, delay);
+    }
+  }, [isOpen, delay]);
+
+  useEffect(() => {
+    if (!isOpen) {
+      setShowContent(false);
+    }
+  }, [isOpen]);
+
+  if (!isOpen || !showContent) {
+    return null;
+  }
+
+  return (
+    <Flex
+      zIndex={100}
+      position="absolute"
+      background={colors.constant.overlay}
```

Comment:
You don't need to get colors from the theme, you can just use "constant.c100" as a prop